OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseBaseActivity.java

```diff
@@ -130,9 +130,13 @@ protected void restore(Bundle savedInstanceState) {
                     snackbarErrorNotification, this) {
                 @Override
                 protected void onResponse(@NonNull final CourseComponent courseComponent) {
-                    courseComponentId = courseComponent.getId();
-                    invalidateOptionsMenu();
-                    onLoadData();
+                    try {
+                        courseComponentId = courseComponent.getId();
+                        invalidateOptionsMenu();
+                        onLoadData();
+                    }catch (NullPointerException e){
+                        e.printStackTrace();
```

Comment:
There should be a flow in case of failure to show some error/info to the user if course is not available.